include/circt/Dialect/Moore/MIRExpressions.td

```diff
@@ -0,0 +1,30 @@
+//===- MIRExpressions.td - Moore MIR expression ops --------*- tablegen -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This describes the ops for Moore MIR expressions.
+//
+//===----------------------------------------------------------------------===//
+
+include "mlir/Interfaces/InferTypeOpInterface.td"
+
+
+def ConstantOp : MIROp<"constant", [NoSideEffect]> {
+  let summary = "A constant value";
+
+  let arguments = (ins I32Attr:$value);
+  let results = (outs MooreRValueType:$result);
+  let assemblyFormat = "$value attr-dict `:` type($result)";
+}
+
+def VariableDeclOp : MIROp<"vardecl", []> {
+  let summary = "A variable declaration";
+
+  let arguments = (ins StrAttr:$name, I32Attr:$init);
+  let results = (outs MooreLValueType:$result);
+  let assemblyFormat = "$name `=` $init attr-dict `:` type($result)";
+}
```

Comment:
Extreme nitpicking: I think variable declarations in SV are statements, not expressions. Maybe this could live in the `MIRStatements.td` file?